semgrep-core/src/matching/Generic_vs_generic.ml

```diff
@@ -2158,6 +2158,9 @@ and m_pattern a b =
       m_ident a1 b1 >>= fun () -> m_id_info a2 b2
   | G.PatLiteral a1, B.PatLiteral b1 -> m_literal a1 b1
   | G.PatType a1, B.PatType b1 -> m_type_ a1 b1
+  | G.PatConstructor (a1, [ G.PatEllipsis _ ]), B.PatConstructor (b1, _)
```

Comment:
First, if you look at the comment at the top of this file (maybe we should improve it), the pattern is the first
parameter (the axxx) and the target source code the second (the bxxxx), so PatEllipsis can't appear in the bxxx.

Second, this will handle just List(...), but what about List(a, ..., b), or List(a, ...), etc. You probably want to use m_list_with_dots and imitate what we do for handling container elements (see m_container_ordered_elements).